src/Compilers/CSharp/Portable/CodeGen/CodeGenerator.cs

```diff
@@ -102,12 +131,17 @@ private enum IndirectReturnState : byte
 
             _boundBody = Optimizer.Optimize(
                 boundBody, 
-                debugFriendly: _optimizations != OptimizationLevel.Release, 
+                debugFriendly: _ilEmitStyle != ILEmitStyle.Release, 
                 stackLocals: out _stackLocals);
 
             _methodBodySyntaxOpt = (method as SourceMethodSymbol)?.BodySyntax;
         }
 
+        private bool IsDebugPlus()
+        {
+            return this._module.Compilation.Options.DebugPlusMode;
```

Comment:
I'd argue that anything that csc.exe can do should be also doable using public Compilation API. We don't need to be too specific about what the difference is between Release and DebugFriendlyRelease. We can document that DebugFriendlyRelease results in slightly better debugging experience while still performing most optimizations.